core/services/relay/evm/tron_transmitter.go

```diff
@@ -0,0 +1,172 @@
+	chain, ok := opts.ConfigWatcher.chain.(legacyevm.ChainTronSupport)
+	if !ok {
+		return nil, fmt.Errorf("chain %s does not support TRON", opts.ConfigWatcher.chain.ID())
+	}
+
+	// Check there is at least one node
+	if len(chain.Nodes()) == 0 {
+		return nil, fmt.Errorf("no nodes found for chain %s", opts.ConfigWatcher.chain.ID())
+	}
+
+	// Use an (extra) write-specific node URL for the Tron client
+	// Notice: TronClient is not multinode aware, so we need to use the first node
+	nodeURL := (*chain.Nodes()[0]).HTTPURLExtraWrite.URL()
+	tronClient, err := tronclient.CreateFullNodeClient(nodeURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Tron client: %w", err)
+	}
+
+	// Start the Tron TXM
+	tronTXM := trontxm.New(lggr, tronkeystore.NewLoopKeystoreAdapter(opts.Keystore), tronClient, trontxm.TronTxmConfig{
+		EnergyMultiplier: 3,
+		StatusChecker:    true,
+	})
+
+	tronEVMTransmitterWrapper := &tronEVMTransmitterWrapper{
+		evmTransmitter: opts.EVMTransmitter,
+	}
+
+	senderAddress, err := opts.Keystore.GetNextAddress(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct the Tron contract transmitter, it's slightly different from the EVM contract transmitter and due to mismatching types we have to apply the transmitter options manually
+	transmitter := tron.NewOCRContractTransmitter(ctx, tronEVMTransmitterWrapper, tronsdk.EVMAddressToAddress(opts.ConfigWatcher.contractAddress), tronsdk.EVMAddressToAddress(senderAddress), tronTXM, lggr).WithEthereumKeystore()
+	transmitterOptions := &transmitterOps{
+		reportToEvmTxMeta: nil,
+		excludeSigs:       false,
+		retention:         0,
+		maxLogsKept:       0,
+	}
+
+	for _, opt := range ocrTransmitterOpts {
+		opt(transmitterOptions)
+	}
+
+	if transmitterOptions.excludeSigs {
+		lggr.Info("Excluding signatures from transmissions")
+		transmitter = transmitter.WithExcludeSignatures()
+	}
+	if transmitterOptions.reportToEvmTxMeta != nil {
+		lggr.Info("Using EVM TxMeta for Tron Transmissions")
+		transmitter = transmitter.WithReportToEthMetadata(transmitterOptions.reportToEvmTxMeta)
+	}
+
+	tronTransmitter := newTronTransmitterWrapper(transmitter, tronTXM)
+
+	// NOTE: Ideally we'd call the Start() method within a service hook, however the commit or exec providers aren't always started unless a new job is created
+	// To protect against this, we start the Tron Transmitter here, however Closing it is handled by the providers themselves
+	if err := tronTransmitter.Start(ctx); err != nil {
+		return nil, err
+	}
+
+	return tronTransmitter, nil
+}
+
+var _ ContractTransmitter = (*TronTransmitterWrapper)(nil)
+
+// Simple wrapper around the tron.ContractTransmitter to provide start / stop hooks to the tron txm
+type TronTransmitterWrapper struct {
+	transmitter tron.ContractTransmitter
+	txm         *trontxm.TronTxm
+}
+
+func newTronTransmitterWrapper(transmitter tron.ContractTransmitter, txm *trontxm.TronTxm) *TronTransmitterWrapper {
+	return &TronTransmitterWrapper{
```

Comment:
1. No need for the impl to be public
2. No need to return a specific type, but a general type

```suggestion
type tronTransmitterWrapper struct {
	transmitter tron.ContractTransmitter
	txm         *trontxm.TronTxm
}

func newTronTransmitterWrapper(transmitter tron.ContractTransmitter, txm *trontxm.TronTxm) *ContractTransmitter {
	return &tronTransmitterWrapper{
```